src/commissaire_http/dispatcher/__init__.py

```diff
@@ -154,11 +155,13 @@ def _get_params(self, environ, route, route_data):
 
         # If we are a PUT or POST look for params in wsgi.input
         if environ['REQUEST_METHOD'] in ('PUT', 'POST'):
-            if environ.get('CONTENT_LENGTH') and environ['CONTENT_LENGTH']:
+            content_length = int(environ.get('CONTENT_LENGTH', 0))
+            if content_length > 0:
                 try:
-                    params.update(json.loads(environ['wsgi.input'].read(
-                        int(environ['CONTENT_LENGTH'])).decode()))
-                except json.decoder.JSONDecodeError as error:
+                    wsgi_input = environ['wsgi.input'].read(content_length)
+                    more_params = eval(wsgi_input.decode())
+                    params.update(more_params)
+                except (ValueError, json.decoder.JSONDecodeError) as error:
```

Comment:
Come to think of it, was that supposed to be JSON?  I just caught it trying to pass a dictionary-like string like `{'a': 123, 'b': 456}` to `params.update()`.